### DISCUSSION TITLE
It renames the file but put the content in another note "untitled"

### DISCUSSION THREAD

**Discussion by davidszerman** [SilentVoid13/Templater#983] [Help] [answered]
Hello,

I do not want to create a bug as a similar code works for other templates.

Here is the code ; 

`---
alias:
created: <% tp.date.now("YYYY-MM-DD HH:mm") %>
---

--- 
<%* let qcExamen = (await tp.system.suggester((item) => item.basename, app.vault.getMarkdownFiles().filter(file => file.path.startsWith("Notes/Sante/Ressources/Examens")), false, "Examen...")).basename
const date = await tp.system.prompt("Date", tp.date.now("YYYY-MM-DD"))
const detail = await tp.system.prompt("Détail (titre)")
tp.file.rename(date+"-"+qcExamen) %>

# <% date %> - <% qcExamen %> - <% detail %>

## Informations

- Date:: [[<% date %>]]
- Examen::  [[<% qcExamen %>]]
- Détail:: <% detail %>

## Notes

<% tp.file.cursor(1) %> 

--- `

When I "run" the code (actually, I create a new note in a specific folder and the templater plugin is set up to apply the template to all notes inside the folder), I have got all the prompts as expected but 

1. It renames correctly the file as expected
2. But it inserts all the content in another note called untitled

Could you please help me to understand what I am missing ?

Thank you

**Comment by Zachatoo** [comment, marked as answer]
I believe you need to await the rename, so that the rename finishes first before proceeding.
```js
await tp.file.rename(date+"-"+qcExamen)
```
